code/parachain/frame/staking-rewards/src/test/mod.rs

```diff
@@ -369,25 +416,27 @@ fn unstake_in_case_of_not_zero_claims_and_early_unlock_should_work() {
 		total_rewards,
 		total_shares,
 		Some(claim),
-		|pool_id, stake_id, unlock_penalty, _, staked_asset_id| {
-			assert_ok!(StakingRewards::unstake(Origin::signed(staker), stake_id));
-			assert_eq!(StakingRewards::stakes(stake_id), None);
+		|pool_id, unlock_penalty, _, staked_asset_id| {
+			let fnft_asset_account = FinancialNft::asset_account(&1, &0);
+
+			assert_ok!(StakingRewards::unstake(Origin::signed(staker), 1, 0));
+			assert_eq!(StakingRewards::stakes(1, 0), None);
 			assert_last_event::<Test, _>(|e| {
 				matches!(
 					e.event,
-					Event::StakingRewards(crate::Event::Unstaked { owner, position_id })
-						if owner == staker && position_id == stake_id
+					Event::StakingRewards(crate::Event::Unstaked { owner, fnft_collection_id, fnft_instance_id })
+						if owner == staker && fnft_collection_id == 1 && fnft_instance_id == 0
 				)
 			});
 
 			let penalty = unlock_penalty.mul_ceil(amount);
 			let claim_with_penalty = (Perbill::one() - unlock_penalty).mul_ceil(claim);
 			let rewards_pool = StakingRewards::pools(pool_id).expect("rewards_pool expected");
 			assert_eq!(balance(staked_asset_id, &staker), amount * 2 - penalty);
-			assert_eq!(
-				balance(staked_asset_id, &StakingRewards::pool_account_id(&pool_id)),
-				amount * 2 + penalty
-			);
+			// NOTE(connor): Used to keep penalty in the pool account, fNFT desing has assets being
```

Comment:
```suggestion
			// NOTE(connor): Used to keep penalty in the pool account, fNFT design has assets being
```

Reply:
It would be nice if we could also assert the fNFT pallet state after this. i.e check if the fNFT no longer exists

Reply:
372dc3cac48dcbd9b8047729c31d6bb6ef3fda33